File: examples/src/Presentations/Conversion/ConvertWholePresentationToHTMLWithMediaFiles.cs
using Aspose.slides.Export


/*
This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Slides for .NET API reference 
when the project is build. Please check https:#docs.nuget.org/consume/nuget-faq for more information. 
If you do not wish to use NuGet, you can manually download Aspose.Slides for .NET API from http:#www.aspose.com/downloads, 
install it and then add its reference to this project. For any issues, questions or suggestions 
please feel free to contact us using http:#www.aspose.com/community/forums/default.aspx
*/

namespace Aspose.slides.Examples.CSharp.Conversion
{
    public class ConvertWholePresentationToHTMLWithMediaFiles
    {
        public static void Run()
        {
            #ExStart:ConvertWholePresentationToHTMLWithMediaFiles
            # The path to the documents directory.
            dataDir = RunExamples.GetDataDir_Conversion()
           const htmlDocumentFileName = "presentationWithVideo.html"
          using (Presentation pres = new Presentation("presentationWith.pptx"))
        {
        VideoPlayerHtmlController controller = new VideoPlayerHtmlController(
        "", htmlDocumentFileName, "http:#www.example.com/")

        HtmlOptions htmlOptions = new HtmlOptions(controller)
        SVGOptions svgOptions = new SVGOptions(controller)

        htmlOptions.HtmlFormatter = HtmlFormatter.CreateCustomFormatter(controller)
        htmlOptions.SlideImageFormat = SlideImageFormat.Svg(svgOptions)

        pres.save(htmlDocumentFileName, SaveFormat.Html, htmlOptions)
}
            }
            #ExEnd:ConvertWholePresentationToHTMLWithMediaFiles
        }
    }

File: examples/src/Presentations/Conversion/ConvertWholePresentationToHTMLWithMediaFiles.py
import aspose.slides as slides


def convert_to_html_with_media():
    #ExStart:ConvertWholePresentationToHTMLWithMediaFiles
    # The path to the documents directory.
    dataDir = "./examples/data/"
    outDir = "./examples/out/"

    htmlDocumentFileName = outDir + "convert_to_html_with_media_out.html"

    with slides.Presentation(dataDir + "presentation_with_media.pptx") as pres:
        controller = slides.export.VideoPlayerHtmlController("", htmlDocumentFileName, "http://www.example.com/")

        htmlOptions = slides.export.HtmlOptions(controller)
        svgOptions = slides.export.SVGOptions(controller)

        htmlOptions.html_formatter = slides.export.HtmlFormatter.create_custom_formatter(controller)
        htmlOptions.slide_image_format = slides.export.SlideImageFormat.svg(svgOptions)

        pres.save(htmlDocumentFileName, slides.export.SaveFormat.HTML, htmlOptions)
    #ExEnd:ConvertWholePresentationToHTMLWithMediaFiles